File: src/01-easy/lumen/solution.dart
import 'dart:io';
import 'dart:math';
void main(){
    int n=int.parse(stdin.readLineSync()!),l=int.parse(stdin.readLineSync()!);
    var g=List.generate(n,(_)=>stdin.readLineSync()!.trim().split(' '));
    int d=0;
    for(int r=0;r<n;r++) for(int c=0;c<n;c++){bool lit=false;
        for(int r2=0;r2<n&&!lit;r2++) for(int c2=0;c2<n&&!lit;c2++)
            if(g[r2][c2]=='C'&&max((r-r2).abs(),(c-c2).abs())<l) lit=true;
        if(!lit) d++;}
    print(d);
}

File: src/01-easy/lumen/solution.rs
use std::io::{self,BufRead};
fn main(){
    let stdin=io::stdin();let mut it=stdin.lock().lines();
    let n:usize=it.next().unwrap().unwrap().trim().parse().unwrap();
    let l:i32=it.next().unwrap().unwrap().trim().parse().unwrap();
    let g:Vec<Vec<String>>=(0..n).map(|_|it.next().unwrap().unwrap().split_whitespace().map(String::from).collect()).collect();
    let mut d=0;
    for r in 0..n{for c in 0..n{
        let lit=(0..n).any(|r2|(0..n).any(|c2|g[r2][c2]=="C"&&((r as i32-r2 as i32).abs().max((c as i32-c2 as i32).abs()))<l));
        if!lit{d+=1;}
    }}
    println!("{}",d);
}
